### ISSUE TITLE
Cross-map travel

### ISSUE THREAD

**Issue by MattJeanes** [MattJeanes/TARDIS#51]
This has been an idea for a long while, but since it would only work in singleplayer I'm not really sure if it's something I want to invest time and effort in. It's low priority anyway.

**Comment by drwills83** [comment]
it would be lethal if this was possible though

**Comment by MattJeanes** [comment]
Been thinking about this a bit more, mostly because I realised most people actually use the tardis in singleplayer rather than multiplayer - so it would be fairly beneficial to people.

I also thought of a way to get this working in multiplayer - only allow it if everyone on the server is in the tardis when it activates and maybe even call a vote which requires everyone to say yes to.

It may even be possible to change the loading screen to some sort of vortex animation but the server might reset it on map load, we'll see.

**Comment by SingularMix** [comment]
The voting is a good idea! What if it worked something like everyone that is in the Tardis has to vote and if everyone in tardis votes "yes" then everyone in tardis teleports to different map and the map changes just for people that were in TARDIS; that way it doesn't effect other players in game. If that is something that is even possible.

**Comment by logoster** [comment]
@SingularMix sadly that wouldn't actually be possible as far as i'm aware, as server's can only run one map at a time, so unless it was programmed to start a new server (which really doesn't sound possible either) that wouldn't really world

would be cool though

**Comment by Noname12345GER** [comment]
Would it be possible to make this possible across servers instead? The destinated Server must allow the Player to Spawn a TARDIS though.

**Comment by benrob0329** [comment]
Perhaps it could be set up on both servers to allow cross travel? Or if running a private (aka friends only) server a simple map change if everyone is in the TARDIS would work.

**Comment by Noname12345GER** [comment]
@benrob0329 good point, but how Does the TARDIS recognize an suitable landing spot since you cant expect a coordinates to be aviable (not out of map). Maybe this can be solved by pre-defined landing spots

**Comment by benrob0329** [comment]
Perhaps you could use a scanner or something to choose a spot to materialize at?

**Comment by MattJeanes** [comment]
I could (ab)use player spawn positions to place the tardis after map change.

**Comment by Noname12345GER** [comment]
@MattJeanes Is Cross-Server travel even possible?

**Comment by MattJeanes** [comment]
should be, dunno how well it would work though

**Comment by benrob0329** [comment]
Is there a way to send messages cross server?

Perhaps you could do something like this:

1. Server A sends a message to server B saying that a TARDIS with X amount of people, and Y configuration would like to join.

2. Server B check for extentions (mostly for interiors) and space (spawning space and number of players) and either confirms or denies joining.

3. If Server B confirms joining, server A serializes objects into a file and sends it to server B, Server B then spawns a demated TARDIS a short distance away from a player spawnpoint (checking for obstructions as well)

4. Server A then sends clients in TARDIS to Server B, spawning them inside the TARDIS at their reletive locations.

**Comment by MattJeanes** [comment]
yea could do, I've written stuff that talks cross server before:

https://github.com/MattJeanes/Node-GMod-Example

**Comment by Noname12345GER** [comment]
I think this should be restrictable because I thought Server Owners won't make such an overpowered machinery aviable for players. You can't Denny it. The TARDIS can be used as a weapon. This might be a criteria if the TARDIS is even added to the server. With might affect the addon's popularity. But im not sure if this thread is focusing on the cross map travel or does it includes the server travel aswell?

**Comment by Noname12345GER** [comment]
What about making a seperate Addon for the map and/or Server travel?

**Comment by Noname12345GER** [comment]
It got pretty silent on the TARDIS GitHub recently...

**Comment by GsTheMann** [comment]
Reviving this thread, this needs to happen, would be amazing.

**Comment by thedoctor1415** [comment]
I can imagine it working slightly like this mod
https://steamcommunity.com/sharedfiles/filedetails/?id=2418016963&searchtext=transitions

**Comment by MattJeanes** [comment]
This would still be pretty cool, but I think a simple way to do it without getting overcomplicated would be to only allow travel between saved destinations on other maps, and have a checkbox to enable showing other maps with a warning that it will change your map.

It will then save a file to disk with the relevant data for cross map travel which the addon will load on map load to check if it needs to spawn in a TARDIS and put the player inside. 

This will also only work in singleplayer.

**Comment by parar020100** [comment]
Multiplayer could be possible, there are "change map" voting systems, for example
